dist/lucca-ui.js

```diff
@@ -872,6 +873,437 @@
+			}
+		};
+
+		/*****************/
+		/***** USERS *****/
+		/*****************/
+
+		var getLimit = function() {
+			var limit = MAGIC_NUMBER_maxUsers;
+
+			if (ctrl.asyncPagination) {
+				limit = MAX_COUNT + 1;
+			}
+			return limit;
+		};
+
+		var getUsersAsync = function(input) {
+			var formerEmployees = "formerEmployees=" + ($scope.showFormerEmployees ? "true" : "false");
+			var limit = "&limit=" + getLimit();
+			var clue = "&clue=" + input;
+			var query = "/api/v3/users/find?" + (init ? "" : (clue + "&")) + formerEmployees + limit;
+			var deferred = $q.defer();
+
+			// Happen when the user starts typing a name, then waits enough to call the api and continues typing
+			// We do not want to treat the result of the previous request since they are now obsolete
+			if (promise) {
+				promise.then(function (response) {}); // do nothing with the results
+			}
+
+			reinitTimeout();
+
+			promise = $http.get(query);
+			promise.then(
+				function (response) {
+					deferred.resolve(response.data.data.items);
+				}, 
+				function (error) {
+					deferred.reject(error); // HANDLE ERROR
+				}
+			);
+			return deferred.promise;
+		};
+
+		/**********************/
+		/***** PAGINATION *****/
+		/**********************/
+
+		var hasPagination = function(users) {
+			if (users.length > MAX_COUNT) {
+				return true;
+			}
+			return false;
+		};
+
+		var handlePaginationAsync = function(input) {
+			var delay = 2500; // default delay is 2,5s
+			var deferred = $q.defer();
+
+			reinitTimeout();
+
+			// launch new timeout 
+			timeout.count = $timeout(function() {
+				getCountAsync(input).then(
+					function (count) {
+						deferred.resolve(count);
+					},
+					function (error) {
+						// HANDLE ERROR
+						deferred.reject(error);
+					}
+				);
+			}, delay);
+			return deferred.promise;
+		};
+
+		var getCountAsync = function(input) {
+			var deferred = $q.defer();
+			var dtContractEnd = "&dtcontractend=since," + moment().format("YYYY-MM-DD") + ",null";
+			var query = "/api/v3/users?name=like," + input + "&fields=collection.count" + ($scope.showFormerEmployees ? "" : dtContractEnd); // query for count
+
+			delete timeout.count;
+			$http.get(query).then(
+				function(response) {
+					deferred.resolve(response.data.data.count);
+				},
+				function(error) {
+					deferred.reject(error);
+				}
+			);
+			return deferred.promise;
+		};
+
+		var addOverflowMessage = function(users, maxNbUsers) {
+			var firstUsers = _.first(users, MAX_COUNT);
+			firstUsers.push({ overflow: MAX_COUNT + "/" + maxNbUsers });
+			return firstUsers;
+		};
+
+		var updateOverflowMessage = function(users, nbUsers, maxNbUsers) {
+			_.last(users).overflow = nbUsers + "/" + maxNbUsers;
+			return users;
+		};
+
+		/********************/
+		/***** HOMONYMS *****/
+		/********************/
+
+		var hasHomonyms = function(users) {
+			var usersWithoutHomonyms = _.uniq(users, function(user) { return (user.firstName + user.lastName); });
```

Comment:
what you use here to check user unicity is not sufficient because
 - it is case sensitive, `BRAD PITT` and `Brad Pitt` are not considered as homonyms
 - it does not take into account french accentuation `jérôme` is the same as `jerome` in french but not here